src/EditModel/SectionDialog.component.js

```diff
@@ -216,40 +306,136 @@ class SectionDialog extends React.Component {
             description: this.state.description,
             showRowTotals: this.state.showRowTotals,
             showColumnTotals: this.state.showColumnTotals,
+            displayOptions: JSON.stringify(this.state.displayOptions),
             disableDataElementAutoGroup: this.state.disableDataElementAutoGroup,
-            dataElements: assignedDataElementStore.state.map(de => ({ id: de })),
+            dataElements: assignedDataElementStore.state.map(de => ({
+                id: de,
+            })),
             indicators: assignedIndicatorStore.state.map(i => ({ id: i })),
-            sortOrder: this.props.sectionModel.sortOrder || modelToEditStore
-                .state
-                .sections
-                .toArray()
-                .reduce((prev, s) => Math.max(prev, s.sortOrder + 1), 0),
+            sortOrder:
+                this.props.sectionModel.sortOrder ||
+                modelToEditStore.state.sections
+                    .toArray()
+                    .reduce((prev, s) => Math.max(prev, s.sortOrder + 1), 0),
         });
-        sectionModel.save()
-            .then((res) => {
-                snackActions.show({ message: this.getTranslation('section_saved') });
-                this.context.d2.models.sections.get(res.response.uid, {
-                    fields: [
-                        ':all,dataElements[id,categoryCombo[id,displayName]]',
-                        'greyedFields[categoryOptionCombo,dataElement]',
-                    ].join(','),
-                })
-                    .then((section) => {
+        sectionModel
+            .save()
+            .then(res => {
+                snackActions.show({
+                    message: this.getTranslation('section_saved'),
+                });
+                this.context.d2.models.sections
+                    .get(res.response.uid, {
+                        fields: [
+                            ':all,dataElements[id,categoryCombo[id,displayName]]',
+                            'greyedFields[categoryOptionCombo,dataElement]',
+                        ].join(','),
+                    })
+                    .then(section => {
                         this.props.onSaveSection(section);
                     });
             })
-            .catch((err) => {
+            .catch(err => {
                 log.warn('Failed to save section:', err);
                 snackActions.show({
                     message: this.getTranslation('failed_to_save_section'),
                     action: this.getTranslation('ok'),
                 });
             });
-    }
+    };
+
+    handleChoosePivotMode = (e, pivotMode) => {
+        const pivotedCategory =
+            pivotMode === 'move_categories'
+                ? this.state.categories[0].id
```

Comment:
I think we should be a bit defensive here - what if there's no categories in the DB, or if the request fails?